src/chart.js

```diff
@@ -8,6 +8,7 @@ Chart.helpers = require('./helpers/index');
 // @todo dispatch these helpers into appropriated helpers/helpers.* file and write unit tests!
 require('./core/core.helpers')(Chart);
 
+Chart._adapters = require('./core/core.adapters');
```

Comment:
Do we have to create a `Chart._adapters` variable? We're requiring the adapter directly in the time scale, so I don't think this is used

Reply:
https://github.com/simonbrunel/chartjs-adapter-luxon/blob/master/src/index.js#L5

Reply:
Could we do that via an import? Maybe something like:

    import { _adapters, helpers } from 'chart.js';

Reply:
I guess we could import that way in the Luxon adapter but it doesn't change the fact we still need to export `_adapter` to be accessed from external adapters. See this line as an equivalent to:

```javascript
export { default as _adapters } from './core/core.adapters';

// or

module.exports._adapters = require('./core/core.adapters');
```

Reply:
Yeah, I'm fine with exporting `_adapter`. I was just wondering if we could avoid adding stuff to the `Chart` object since it seems like the legacy way of doing things now that we're trying to make everything importable

Reply:
> Yeah, I'm fine with exporting _adapter

What's your suggestion to avoid exporting `_adapter`?

Reply:
I'm not suggesting to avoid exporting `_adapter`. I'm wondering if we should instead use one of the ways you mentioned in your last comment since they seem like a more modern style of doing this:

```
export { default as _adapters } from './core/core.adapters';

// or

module.exports._adapters = require('./core/core.adapters');
```

Reply:
> I'm wondering if we should instead use one of the ways you mentioned in your last comment

It's already what we are doing with [`module.exports = Chart;`](/chartjs/Chart.js/blob/master/src/chart.js#L43).

Reply:
The luxon adapter right now is importing the entire chart.js library. Can it import just the modules that it needs instead?

I imagine that importing the whole library in the luxon adapter will stop users who only want to import the time chart from being able to tree shake out the unused parts

Reply:
> Do we have to create a Chart._adapters variable?

Answering your initial question: **yes**, else we can't import / use `_adapters` from external code, whatever the import method (global, CJS, ESM, etc.). Right now, we gather all our exports in the `Chart` [local variable](/chartjs/Chart.js/blob/master/src/chart.js#L4) and [export this object at the end](/chartjs/Chart.js/blob/master/src/chart.js#L43) instead of duplicating `module.exports.*` at every lines.

>  ... it seems like the legacy way of doing things ...

It's only applicable to the code living in the core repository because we don't want to broadcast the `Chart` namespace in all our sources. Now, only `src/chart.js` knows about this global [`Chart` namespace](/chartjs/Chart.js/blob/master/src/chart.js#L45) (which is kept for backward compatiblity), every other `Chart.` access in this file is about the local (exported) variable.

> The luxon adapter right now is importing the entire chart.js

That's not true because it uses `chart.js` as [external dependency](/simonbrunel/chartjs-adapter-luxon/blob/master/rollup.config.js#L25). That means the UMD build contains absolutely no code from the `chart.js` library. 

> Can it import just the modules that it needs instead?

It could, but that changes nothing for UMD consumers because it will [still `require('chart.js')`](https://gist.github.com/simonbrunel/628ca3b62bcc4676ae70b6eeb4d53374#file-partial-imports-umd-js-L2) (see this [gist of the generated UMD builds](https://gist.github.com/simonbrunel/628ca3b62bcc4676ae70b6eeb4d53374). It could actually be less optimized doing this way since it doesn't alias `helpers`. Also, there is an [additional 'default' logic](https://gist.github.com/simonbrunel/628ca3b62bcc4676ae70b6eeb4d53374#file-full-import-umd-js-L7) which may not work correctly with partial imports.

> ... from being able to tree shake out the unused parts

Does `chart.js` support tree shaking? I'm not even sure CJS support tree shaking when importing another CJS module (e.g. `var helpers = require('chart.js').helpers;`).

Reply:
Perhaps my suggestion would only help ES6 users and not CJS/UMD users. I still think it would be worth doing in that case. But since the API is private, we can always change it later, so I'm okay not worrying about it for now. We'll be revisiting how to make moment optional in a separate PR already anyway and this concern seems secondary to that one